edx_solutions_api_integration/users/tests.py

```diff
@@ -2286,3 +2288,14 @@ def test_users_courses_detail_post_missing_child_content_id(self):
 
         response = self.do_post(test_uri, data=position_data)
         self.assertEqual(response.status_code, 400)
+
+    def test_users_courses_grades_detail_race_condition(self):
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
```

Comment:
Could you please document the purpose of this test case and also how it achieved the goals to write it?